cdap-client/src/main/java/co/cask/cdap/client/ServiceClient.java

```diff
@@ -83,6 +85,30 @@ public ServiceSpecification get(Id.Service service)
   }
 
   /**
+   * Gets a {@link ServiceSpecification} for a {@link Service}.
+   *
+   * @param service ID of the service
+   * @return {@link ServiceSpecification} representing the service
+   * @throws IOException if a network error occurred
+   * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
+   * @throws NotFoundException if the app or service could not be found
+   */
+  public ServiceSpecification get(ProgramId service)
+    throws IOException, UnauthenticatedException, NotFoundException, UnauthorizedException {
+
+    URL url = config.resolveNamespacedURLV3(service.getNamespaceId(),
```

Comment:
Is it possible to use this new API to get the ServiceSpecification of a service deployed via the older (non-versioned) APIs?